File: algorithms/075_sort_colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int left=0, right=nums.size()-1;
        while (left<right) {
            while (nums[left]<2 && left <= right) left++;
            while (nums[right]==2 && left <= right) right--;
            if (left < right) {swap(nums, left, right);
            left++;
            right--;}
        }
        left=0;
        while (left < right ) {
            while (nums[left]==0 && left <= right) left++;
            while (nums[right]>=1 && left<=right) right--;
            if (left<right) {swap (nums, left, right);
            left++;
            right--;}
        }
        
    }
    void swap (vector<int>& nums, int left, int right) {
        int temp=nums[right];
        nums[right]=nums[left];
        nums[left]=temp;
    }
};

File: algorithms/python3/075_sort_colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        left,right,cur=0,len(nums)-1,0
        while cur<=right:
            if nums[cur]==2:
                nums[cur],nums[right]=nums[right],nums[cur]
                right-=1
            elif nums[cur]==0:
                nums[cur],nums[left]=nums[left],nums[cur]
                cur+=1
                left+=1
            else:
                cur+=1

a=Solution()

nums=[2,0,2,1,1,0]

a.sortColors(nums)

